### PR TITLE
Add two new settings: Pan camera with middle click and Immediate delete mode

### PR DESCRIPTION
**Pull request by DavidTriphon**
Originally mentioned in Issue #466.

Panning camera with the middle mouse button is what it says on the tin. Other actions still occur on left click, and you can no longer pan the camera with left click when no building is selected. This allows the user to pan the camera with the mouse without having to deselect buildings.

Immediate delete mode allows the user to delete buildings without deselecting the building first. This setting cooperates with the first setting mentioned here. When active, right clicking will always delete what is under the mouse cursor. right clicking will not deselect the current building. The other ways to deselect a building by clicking on the active symbol at the bottom of the screen or pressing the relevant number key still work for deselection and does not make camera panning with the left mouse button impossible.

Also fixed a bug regarding mouseUp events that weren't paying attention to whether the relevant mouse button was being lifted or not.

There are a few weird conflicts that result from holding or pressing multiple mouse buttons at the same time when these settings are activated. But only happen for as long as multiple buttons are pressed and always resolve themselves once the buttons are released. These may or may not need to be fixed.

### CODE REVIEW

File: src/js/game/camera.js [hunk 1]
```diff
@@ -653,28 +673,36 @@ export class Camera extends BasicSerializableObject {
...
 
-        this.touchPostMoveVelocity = new Vector(0, 0);
-        this.currentlyMoving = true;
-        this.lastMovingPosition = pos;
-        this.lastMovingPositionLastTick = null;
-        this.numTicksStandingStill = 0;
-        this.didMoveSinceTouchStart = false;
+        if (
+            buttonKey ===
+            (this.root.app.settings.getSetting("middleMousePan")
```
Comment (tobspr) [inline src/js/game/camera.js:695]: Could you make this a getter?

E:g.

`get mousePanButton() { return ... }`

And then

`if (buttonKey === this.mousePanButton)`

Comment (DavidTriphon) [reply src/js/game/camera.js]: I've made the requested changes.

**Comment by DavidTriphon** [comment]
I made the requested changes but I am still seeing the status of the review as "1 change requested" despite that I resolved the conversation. Is there some action I am supposed to do in order to mark that as resolved as well  so that you know it's time to review it again or do I just wait until you confirm that it satisfies your request?

**Comment by tobspr** [comment]
> I made the requested changes but I am still seeing the status of the review as "1 change requested" despite that I resolved the conversation. Is there some action I am supposed to do in order to mark that as resolved as well so that you know it's time to review it again or do I just wait until you confirm that it satisfies your request?

Nah, I always get a notification, I just didn't get to it yet :) Re-requesting the review was the right thing, I just can't check the PR's every day which is why it might take some time.

The PR looks good now and I'll try to test and merge it soon!

**Comment by DavidTriphon** [comment]
Good to know. I'm glad that everything meets your standard of quality. I'll be less impatient in the future; I was worried that I might not have set the right automated processes in motion that would have allowed you to know which PRs to check back on and which to ignore. Now I know so I don't have to worry.

**Comment by bcmpinc** [comment]
This also implements feature request #583.

**Comment by tobspr** [comment]
I already added immediate delete mode in the meantime, could you have a look at the conflicts? :)

**Comment by DavidTriphon** [comment]
Will do. Sorry about the delay, it's been a busy week.

**Comment by tobspr** [comment]
no worries, I'll mark it as changes required until then :)

**Comment by Polarbub** [comment]
Is this still being worked on?

**Comment by DavidTriphon** [comment]
Sorry, school has kept me very occupied. I wanted to get this done a month ago but I ran into the new GitHub permissions workflow thing and that stopped me then. I've fixed it since then. Hopefully it should be a breeze for me to fix the conflicts. I know immediate delete mode was implemented simultaneously in another commit so that will probably take the most effort.

**Comment by DavidTriphon** [comment]
May I just say one thing:
Thank god for jetbrains conflict resolution interface and the magic wand tool for non-conflicting changes.

**Comment by DavidTriphon** [comment]
Looking at the yaml lint test details shows errors and warnings in the lang files that I didn't touch, so I assume they are in master as well.

**Comment by DavidTriphon** [comment]
Currently looking into why 3 of the 5 files still believe the entire files needs to be diffed. I think it's possible that I reverted some things. If I can't tell if I did or not that's bad.

**Comment by DavidTriphon** [comment]
I can't figure it out. Looking through history for camera.js shows that the line endings have always been CRLF. So I don't think it's that. And in both my PR and master, the file uses spaces and not tabs, so it's not that either. I'm not sure why it thinks the entire file changed. I'm not comfortable merging this until that's fixed. I'm all ears to anyone who can point it out.

**Comment by bcmpinc** [comment]
Iirc git by default does automatic CRLF to LF line ending conversion upon commit. 

In any case, you can check the differences here on github while ignoring whitespace:
image

Incidentally the yaml lint errors are caused due to tobspr changing the line endings in one of those yaml files from LF to CRLF.

**Comment by tobspr** [comment]
I'll have a look soon! Would be nice to have the whitespace changes fixed tho

**Comment by DavidTriphon** [comment]
Actually, I’m not sure this PR is ready besides whitespace. I played the released version and realized that my merge commits got rid of a feature you had that I didn’t, right clicking to deselect when their is no building to delete. This or gets rid of that. That should not be gotten rid of.

I’ve been thinking about what benefits are left in this PR that are worthwhile. The only thing I can think of is middle button pan, but I’m not entirely sure that should be a setting. In all honesty, I think camera pan would be better as a keyboard than a left click or middle click setting. Some people might prefer right click camera pan for all I know. In which case I think that possibly this PR should be canned and a new one made to allow rebindable mouse keys for mouse key functions like place, delete, deselect, camera pan, etc.

**Comment by tobspr** [comment]
The middle moue button pan is nice to have and on my todo list.

I'll add it to a trello card and close this pr, but it's on my mind so I might check it out when implementing the feature

Thanks a lot though!

**Pull request closed without merging**